src/lxc/lxccontainer.c

```diff
@@ -4659,6 +4680,152 @@ static bool do_lxcapi_restore(struct lxc_container *c, char *directory, bool ver
 
 WRAP_API_2(bool, lxcapi_restore, char *, bool)
 
+static int do_lxcapi_mount(struct lxc_container *c,
+			const char *source, const char *target,
+			const char *filesystemtype, unsigned long mountflags,
+			const void *data, struct lxc_mount *mnt) {
+	int ret = -1, len, fd, status;
+	char *template, *suff, *path;
+	pid_t pid, init_pid;
+	struct stat sb;
+
+	len = strlen(c->lxc_conf->lxc_shmount.path_host) + (sizeof("/lxcmountXXXXXX") - 1);
+	template = malloc(len);
```

Comment:
You need `malloc(len + 1)` for the terminating `\0` byte. This also ensures that the check
    
    ret = snprintf();
    if (ret < 0 || ret >= len)

below makes sense.

Reply:
If I get everything right, just not subtracting 1 above should be enough (both times)